timemachine/fe/free_energy.py

```diff
@@ -120,6 +120,10 @@ class InitialState:
     ligand_idxs: NDArray
     protein_idxs: NDArray
 
+    def __post_init__(self):
+        assert self.ligand_idxs.dtype == np.int32 or self.ligand_idxs.dtype == np.int64
```

Comment:
Might also be good to add

`assert self.ligand_idxs.dtype == self.protein_idxs.dtype`

Just to be sure there isn't inconsistency between the two

Reply:
This matters less - as long as both are int types, a concatenation will promote to the most precise value.